test/rekt/resources/channel_impl/channel_impl.yaml

```diff
@@ -17,6 +17,12 @@ kind: {{ .kind }}
 metadata:
   name: {{ .name }}
   namespace: {{ .namespace }}
+  annotations:
```

Comment:
when looking at the reconciler tests, I never saw the `messaging.knative.dev/async-handler` annotation, hence adding this